src/resources.js

```diff
@@ -185,8 +176,22 @@ export class Resources {
       this.checkPendingChangeHeight_(element);
     });
 
+    // WAT, Why is this set to true here and false in the same method above?
     this.relayoutAll_ = true;
-    this.schedulePass();
+    // Ensure that we attempt to rebuild things when DOM is ready.
+    let docReady = false;
+    this.docState_.onReady(() => {
+      this.documentReady_ = true;
+      this.forceBuild_ = true;
+      this.relayoutAll_ = true;
+      this.schedulePass();
+      this.monitorInput_();
+      docReady = true;
+    });
+
+    if (!docReady) {
+      this.schedulePass();
```

Comment:
Is this specifically because of microtasks? According to the Pass API calling schedulePass twice should not schedule more than one task. I'd rather see if schedulePass working correctly then working around in the clients.